packages/ckeditor5-dev-build-tools/src/plugins/cssStyles.ts

```diff
@@ -0,0 +1,305 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { createFilter } from '@rollup/pluginutils';
+import { parse, type Rule, type Declaration, type Stylesheet } from 'css';
+import type { Plugin, GetModuleInfo, OutputBundle, OutputChunk, NormalizedOutputOptions } from 'rollup';
+
+/**
+ * Filter files only with `css` extension.
+ */
+const filter = createFilter( [ '**/*.css' ] );
+
+export function cssStyles(): Plugin {
+	const styles: Record<string, string> = {};
+
+	return {
+		name: 'ck5-styles',
+		transform( code, id ) {
+			if ( !filter( id ) ) {
+				return;
+			}
+
+			styles[ id ] = code;
+
+			return '';
+		},
+		// ////
+		// `moduleParsed` is just for debugging.
+		// ////
+		moduleParsed( moduleInfo ) {
+			console.log( `Module ${ moduleInfo.id } has been parsed.` );
+		},
+		async generateBundle( output: NormalizedOutputOptions, bundle: OutputBundle ) {
+			const banner = await getBanner( output, bundle );
+			const ids: Set<string> = new Set();
+
+			// Determine import order of files
+			for ( const file in bundle ) {
+				const outputChunk = bundle[ file ]! as OutputChunk;
+				const root = outputChunk.facadeModuleId!;
+				const modules = getCSSModules( root, this.getModuleInfo );
+
+				modules.forEach( id => ids.add( id as string ) );
+			}
+
+			const cssStylesheet = combineStylesheetsIntoOne( ids, styles );
+			const parsedCss = parse( cssStylesheet );
+			const { editorStylesContent, editingViewStylesContent, allStylesContent } = divideStylesheetDependingOnItsPurpose( parsedCss );
+
+			// Emit styles to files
+			this.emitFile( {
+				type: 'asset',
+				fileName: 'styles.css',
+				source: unifyFileContentOutput( allStylesContent, banner )
+			} );
+			this.emitFile( {
+				type: 'asset',
+				fileName: 'editor-styles.css',
+				source: unifyFileContentOutput( editorStylesContent, banner )
+			} );
+			this.emitFile( {
+				type: 'asset',
+				fileName: 'content-styles.css',
+				source: unifyFileContentOutput( editingViewStylesContent, banner )
+			} );
+		}
+	};
+}
+
+/**
+ * @param content is a CSS file content.
+ * @param banner
+ */
+function unifyFileContentOutput( content: string | undefined, banner: string ): string {
+	return `${ banner }\n${ content ? content : '' }\n`;
```

Comment:
We can use `MagicString.prepend` to achieve this while still maintaining proper source maps (since string concatenation breaks them).

Reply:
Adresed in 4cb66cf.